wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/WireTextField.swift

```diff
@@ -96,18 +97,60 @@ class WireTextField: UITextField {
     }
 
     private func setupTextFieldEvents() {
-        addTarget(self, action: #selector(textFieldDidStartEditing), for: .editingDidBegin)
-        addTarget(self, action: #selector(textFieldDidFinishEditing), for: .editingDidEnd)
+        addAction(UIAction(handler: { [weak self] _ in
+            self?.textFieldDidStartEditing()
+        }), for: .editingDidBegin)
+
+        addAction(UIAction(handler: { [weak self] _ in
+            self?.textFieldDidFinishEditing()
+        }), for: .editingDidEnd)
+
+        addAction(UIAction(handler: { [weak self] _ in
+            self?.updateClearButtonVisibility()
+        }), for: .editingChanged)
+    }
+
+    // MARK: - Custom Clear Button
+
+    private func setupCustomClearButton() {
+        let clearButton = UIButton(type: .custom)
+        let clearImage = UIImage(named: "Clear")?.withRenderingMode(.alwaysTemplate)
```

Comment:
Can't we use SF Symbols? I thought the design team actually wants us to use SF Symbols.
`x.circle.fill`

Reply:
`x.circle.fill` is not the same as the designs. In this case I imported the SVG. I need to get some clarification, on this. For now I will leave it as is and I will come back to you with a resolution on that.